celery/concurrency/thread.py

```diff
@@ -1,45 +1,67 @@
 """Thread execution pool."""
+from __future__ import annotations
 
-from concurrent.futures import ThreadPoolExecutor, wait
+from concurrent.futures import Future, ThreadPoolExecutor, wait
+from typing import TYPE_CHECKING, Any, Callable
 
 from .base import BasePool, apply_target
 
 __all__ = ('TaskPool',)
 
+if TYPE_CHECKING:
```

Comment:
Everything inside this block should have been ignored by codecov, according to https://github.com/celery/celery/blob/651095e3602756237920f4fa7ac170e1322c1939/pyproject.toml#L25

Why isn't it?

Reply:
the config is of pre mypy era of celery, so we can adjust that

Reply:
I don't know what needs to be adjusted on codecov. :eyes:

Reply:
btw codecov is buggy

Reply:
Do you want me to open a PR removing it, and calculate the coverage exclusively with `coverage`?

Reply:
I saw some projects moved to that, you can come with that